src/backend/distributed/utils/node_metadata.c

```diff
@@ -364,12 +364,21 @@ RemoveNodeFromCluster(char *nodeName, int32 nodePort, bool forceRemove)
 	bool hasShardPlacements = false;
 	WorkerNode *workerNode = NULL;
 	List *referenceTableList = NIL;
+	uint32 deletedNodeId = 0;
 
 	EnsureSchemaNode();
 	EnsureSuperUser();
 
 	workerNode = FindWorkerNode(nodeName, nodePort);
 
+	if (workerNode == NULL)
```

Comment:
I'd prefer the following in order not to add the same error message that we'd expect from `DeleteNodeRow()`
```C
if (workerNode != NULL)
{
  deleteNodeId = worker->nodeId;
}
```

In case of non-valid node, `DeleteNodeRow()` is already going to throw the same error.
(Sure, this ends up with Un-doing changes in `DeleteNodeRow` where the error might be useful for other purposes as well.)